File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_reader :last_number

  def initialize(last_number)
    @last_number = last_number
  end

  def square_of_sum
    (1..last_number).sum**2
  end

  def sum_of_squares
    (1..last_number).map { |number| number**2 }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(n) {
    this.n = n
  }

  get sumOfSquares() {
    return [...Array(this.n + 1).keys()].reduce((sum, number) => sum + number**2, 0)
  }

  get squareOfSum() {
    return [...Array(this.n + 1).keys()].reduce((sum, number) => sum + number, 0)**2
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares
  }
}
